sale_order_type/models/sale.py

```diff
@@ -74,6 +75,19 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    @api.constrains("company_id")
+    def _check_so_type_company(self):
+        if self.filtered(
+            lambda r: r.type_id.company_id
+            and r.company_id
+            and r.type_id.company_id != r.company_id
+        ):
+            raise ValidationError(_("Document's company and type's company mismatch"))
```

Comment:
```suggestion
```

If you use `check_company=True`, this is not needed.

https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L3132